shared/logic/BuildingLogic.ts

```diff
@@ -623,10 +623,11 @@ export function applyToAllBuildings<T extends IBuildingData>(
    });
 }
 
-export function getMarketSellAmount(xy: Tile, gs: GameState): number {
+export function getMarketSellAmount(xy: Tile, gs: GameState, maxCapacity = false): number {
```

Comment:
I am not sure whether `maxCapacity` belongs here. Maybe it's better for the caller to figure out whether to apply capacity or not. In general, most functions in BuildingLogic are capacity-agnostic - they are not aware of the capacity settings in the building.

Reply:
This is the function before any change.   Which uses capacity from the building.

```
export function getMarketSellAmount(xy: Tile, gs: GameState): number {
   const building = gs.tiles.get(xy)?.building;
   if (!building) return 0;
   return building.capacity * building.level * totalMultiplierFor(xy, "output", 1, gs);
}
```

Reply:
Yes, the function before is probably bad. Don't worry about this one - I can clean this up and send it for your review.